### DISCUSSION TITLE
Contour plot of AEP

### DISCUSSION THREAD

**Discussion by Teddybear0227** [NatLabRockies/floris#768] [Q&A] [answered]
I am trying to search for a way to plot a contour diagram visualising the percentage loss of AEP for each turbine in wind farm as shown in the attached image (retrieved from https://iopscience.iop.org/article/10.1088/1742-6596/2265/2/022049/pdf ). 

I am thinking of calculating the percentage loss of AEP for each turbine in the following process.
1. using fi.get_turbine_power(), calculate the power produced by each turbine in wind farm
2. multiply the frequency for each wind direction and wind speed, and multiply it by 365.25*24 to get AEP for each turbine

Is there any way to convert such array of AEP into a contour plot. 
<img width="314" alt="Screenshot 2023-12-18 at 12 04 33" src="https://github.com/NREL/floris/assets/85109987/ccb1c58d-c768-4458-979d-c156d3f9c0ba">

Sincerely,

**Comment by misi9170** [comment, marked as answer]
Hi @Teddybear0227, we don't have any inbuilt functionality for generating this kind of plot. However, I think you should be able to achieve what you want pretty quickly with [matplotlib.pyplot.scatter](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.scatter.html). 

Here's an example that could get you started:
```
import matplotlib.pyplot as plt
import numpy as np

x = np.array([1, 2, 3, 4, 1, 2, 3, 4])
y = np.array([0, 0, 0, 0, 1, 1, 1, 1])
aep = np.array([100, 90, 80, 70, 90, 80, 70, 60])

fig, ax = plt.subplots(1,1)
sc = ax.scatter(x, y, c=aep, cmap="cividis", vmin=50, vmax=110)
plt.colorbar(sc)

ax.set_xlim([0, 5])
ax.set_ylim([-0.5, 1.5])

plt.show()
```

which generates

![Example scatter plot with color](https://github.com/NREL/floris/assets/39596329/7c7c8520-4c52-42b1-84b1-46ae22e2bd1a)

In fact, you'll find some very similar code in example 29_floating_vs_fixedbottom_farm.py, so you could also take a look at that for some more refined plotting code.

**Comment by Teddybear0227** [comment]
Hello @misi9170 !! Thank you very much for the information. The method worked perfectly for me.

Sincerely,
